viewing_party/party.py

```diff
@@ -1,23 +1,185 @@
+    average_rating = sum_rating/count
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    watched_movies = user_data["watched"]
+    genre_dict = {}
+
+    #In case when watched_movies is an empty list
+    if watched_movies == []:
+        return None
+    # Iterate through list of dicts called watched_movies
+    else:
+        for movie in watched_movies:
+            if movie["genre"] not in genre_dict:
+                genre_dict[movie["genre"]] = 1
+            else:
+                genre_dict[movie["genre"]] += 1
+
+    #Get the key with the maximum value 
+    most_watched_genre_count = 0
+    most_watched_genre = ""
+
+    # Iterate through list of dicts called genre_dict
+    for genre in genre_dict.keys():
+        if genre_dict[genre] > most_watched_genre_count:
+            most_watched_genre_count = genre_dict[genre]
+            most_watched_genre = genre
+    return most_watched_genre
 
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
-        
-# -----------------------------------------
+def get_unique_watched(user_data):
+    user_watched_list = user_data["watched"]
+    friends_list = user_data["friends"]
+    
+    friends_watched_set = set()
+    unique_watched = []
+
+    #Iterate through the movies in nested friends_list
+    for friend in friends_list:
+        for movie in friend["watched"]:  
+            if movie in user_watched_list:
+                friends_watched_set.add(movie["title"])
+
+    #Iterate through the movies in user_watched_list
+    for movie in user_watched_list:
+        if movie["title"] not in friends_watched_set:
+            unique_watched.append(movie)
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = user_data["watched"]
+    friends_list = user_data["friends"]
+    
+    user_watched_set = set()
+    unique_watched = []
+
+    #Get set of movie titles watched by user
+    for movie in user_watched_list:
+        user_watched_set.add(movie["title"])
+
+    #Iterate through each friend in friends_list
+    for friend in friends_list:
+        #For every movie dict in friend dict
+        for movie in friend["watched"]: 
+            #If title is not in user_watched_set AND title is not in unique_watched_set:
+            if movie["title"] not in user_watched_set and movie not in unique_watched:
+                unique_watched.append(movie)
+    return unique_watched
+    
+
+
 # ------------- WAVE 4 --------------------
-# -----------------------------------------
 
-# -----------------------------------------
+def get_available_recs(user_data):
+    unique_movie = get_friends_unique_watched(user_data)
+    recommended_movies = []
+    
+    for movie in unique_movie:
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+    return recommended_movies

```

Comment:
Great work on keeping your code DRY!
```suggestion
    unique_movie = get_friends_unique_watched(user_data)
    recommended_movies = []
    
    for movie in unique_movie:
        if movie["host"] in user_data["subscriptions"]:
            recommended_movies.append(movie)
            
    return recommended_movies

```